Bowling/Bowling/Bowling.cs

```diff
@@ -0,0 +1,105 @@
+        {
+            //create screen banner
+            createBanner();
+            
+            //get bowling marks from user
+            string bowlingMarks = getBowlingMarks();
+
+            // check to make sure we have values or end calculation
+            if(!bowlingMarks.Equals("0"))
+            {
+                Frames frames = new Frames();
+                
+                // use input string to create a list containing each bowling frames
+                List<Frames> allFrames = frames.CreateFrames(bowlingMarks);
+
+                // the scores for each frame now need to be updated to reflect any additional points for strikes and spares.
+                allFrames = frames.UpdateFrameScores(allFrames);
+
+                // calculate grand total
+                int grandTotal = calculateGrandTotal(allFrames);
+
+                // display grand total
+                Console.WriteLine(String.Concat("Final Bowling Score is : ", grandTotal));
+                Console.WriteLine();
+                Console.WriteLine("Press Any Key To Close This Window...");
+                Console.ReadLine();
+            }
+            else  // end game for no input being given
+            {
+                Console.WriteLine("Without any marks no score can be calculated.  You get a big goose egg!");
+                Console.WriteLine();
+                Console.WriteLine("Press Any Key To Close This Window...");
+                Console.ReadLine();
+            }
+
+        }
+
+        private string getBowlingMarks()
+        {
+            /** *****get input from user for bowling marks.
+                     assumption:  we will get a string with 10 frames and not do any
+                                  validating as per instructions (unless we get an empty string).***** **/
+            
+            Console.WriteLine("Please enter your marks for all 10 bowling frames.  (X = STRIKE   / = SPARE   - = ZERO)");
+            Console.WriteLine("Scores can be entered as one continuous string or with spaces");
+            Console.WriteLine("(ex: XXXXXXXXXXXX OR 5/ 5/ X 5/ 52 5/ 5/ 54 5/ 5/5)");
+            string bowlingMarks = Console.ReadLine();
+
+            bowlingMarks = Regex.Replace(bowlingMarks, @"\s+", String.Empty);  // remove any blank spaces to make a continuous string
+            bowlingMarks = bowlingMarks.ToUpper();  //  make any lower case strikes upper case for easier reading and validation
+
+            // return zero and end game calculation if no input exists
+            if (bowlingMarks.Length == 0)
+            {
+                bowlingMarks = "0";
+            }
+
+            return bowlingMarks;
+        }
+
+        private int calculateGrandTotal(List<Frames> framesCollection)
+        {
+            int grandTotal = 0;
+
+            // iterate through each frame and grab the frameScore totals for each frame and create a grand total
+            for(int i=0; i < frameNumberTotal; i++)
```

Comment:
This looks like a good candidate to use a [foreach loop](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/foreach-in). This is just syntactic sugar around what you've done here, but generally makes the code more clear.

Alternatively, have a look at the LINQ function [Aggregate](https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.aggregate?view=netcore-3.1)